(VDB-1732) Add script to delete header

rmulhol opened the pull request:
- To remove headers that enter the DB and survive past the validation
  window in headerSync (which would otherwise clear out non-canonical
  headers).
- Takes a block number and only executes a delete if a header with
  that block number exists in the DB.

Commit:
(VDB-1732) Add script to delete header

- To remove headers that enter the DB and survive past the validation
  window in headerSync (which would otherwise clear out non-canonical
  headers).
- Takes a block number and only executes a delete if a header with
  that block number exists in the DB.

rmulhol commented on go.mod:
```diff
@@ -5,7 +5,6 @@ go 1.15
 require (
 	github.com/dave/jennifer v1.3.0
 	github.com/ethereum/go-ethereum v1.9.22
-	github.com/hashicorp/golang-lru v0.5.4
```
A little suspicious of modifying imports here, but I couldn't run the tests without adding Clickhouse. Then running `go mod tidy` fixed that up, but also got rid of this 🤔 

Figure maybe we bite the bullet now and restore this import if it turns out it's somehow necessary for plugin compatibility (and/or fix the issue downstream)

paytonrules replied:
Go for it. If there's a plugin compatibility problem that should really be addressed at the plugin.

ecmendenhall reviewed (approved):
Does what it says on the box!

paytonrules commented on cmd/deleteHeader.go:
```diff
@@ -0,0 +1,44 @@
+}
+
+func init() {
+	deleteHeaderCmd.Flags().Int64VarP(&deleteHeaderBlockNumber, "block-number", "b", -1, "block number of the header to delete")
```
Can't you make this required so you don't have to check this later?

paytonrules commented on pkg/datastore/postgres/repositories/header_repository_test.go:
```diff
@@ -273,6 +274,30 @@ var _ = Describe("Block header repository", func() {
+			err := repo.DeleteHeader(blockNumber)
+
+			Expect(err).NotTo(HaveOccurred())
+			persistedHeader, getErr := repo.GetHeaderByBlockNumber(blockNumber)
```
Should GetHeaderByBlockNumber also return postgres.ErrHeaderDoesNotExist ?

paytonrules reviewed (approved):
Couple minor comments.

Commit:
Address PR feedback

- mark block number for deleteHeader as required
- modify GetHeaderByBlockNumber to return postgres.ErrHeaderDoesNotExist
  instead of sql.ErrNoRows if header not found for block number
- update storage watcher to gracefully handle the new error
